### PR TITLE
Add `ignoreImports` option to `import-index` rule

### PR DESCRIPTION
**Pull request by maximelkin**
closes https://github.com/sindresorhus/eslint-plugin-unicorn/issues/307

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#307: import-index: Option to ignore import/export statements](https://issuehunt.io/repos/55832243/issues/307)
---

IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### CODE REVIEW

File: rules/import-index.js [hunk 1]
```diff
@@ -16,19 +16,37 @@ const importIndex = (context, node, argument) => {
...
+
+	return rules;
 };
 
+const schema = [{
+	type: 'object',
+	properties: {
+		ignoreImports: {
+			type: 'boolean'
+		}
```
Comment (fisker) [inline rules/import-index.js]: not familiar with this but is default value needed?

File: test/import-index.js [hunk 2]
```diff
@@ -16,9 +16,17 @@ const error = {
...
 
+const ignoreImportsOptions = [{
+	ignoreImports: true
+}];
+
 ruleTester.run('import-index', rule, {
 	valid: [
 		'const m = require(\'.\')',
+		{
+			code: 'const m = require(\'.\')',
```
Comment (fisker) [inline test/import-index.js]: you might mean to  test `require(./index)`

File: rules/import-index.js (hunk 1 above)
Comment (maximelkin) [reply rules/import-index.js]: done

File: test/import-index.js (hunk 2 above)
Comment (maximelkin) [reply test/import-index.js]: done

File: docs/rules/import-index.md [hunk 3]
```diff
@@ -61,3 +61,25 @@ const m = require('@foo/bar');
 ```js
 import m from '.';
 ```
+
+
+## Options
+
+### ignoreImports
+
+Type: `boolean`\
```
Comment (fisker) [inline docs/rules/import-index.md]:
```suggestion
Type: `boolean`
```

Comment (yakov116) [reply docs/rules/import-index.md]: `\` = `br`

**Pull request merged**